code/game/turfs/simulated/walls_reinforced.dm

```diff
@@ -33,249 +59,265 @@
+			"<span class='notice'>You burn the fungi away with \the [W].</span>")
+			playsound(src, 'sound/items/Welder.ogg', 10, 1)
+			for(var/obj/effect/E in src) //WHYYYY
+				if(E.name == "Wallrot") //WHYYYYYYYYY
+					qdel(E)
+			rotting = 0
+			return
+		if(istype(W,/obj/item/weapon/soap))
+			user.visible_message("<span class='notice'>[user] forcefully scrubs the fungi away with \his [W].</span>", \
+			"<span class='notice'>You forcefully scrub the fungi away with your [W].</span>")
+			for(var/obj/effect/E in src)
+				if(E.name == "Wallrot")
+					qdel(E)
+			rotting = 0
+			return
 		else if(!W.is_sharp() && W.force >= 10 || W.force >= 20)
-			user << "<span class='notice'>\The [src] crumbles away under the force of your [W.name].</span>"
+			user.visible_message("<span class='warning'>With one strong swing, [user] destroys the rotting [src] with his [W].</span>", \
+			"<span class='notice'>With one strong swing, the rotting [src] crumbles away under your [W].</span>")
 			src.dismantle_wall()
 			return
 
 	//THERMITE related stuff. Calls src.thermitemelt() which handles melting simulated walls and the relevant effects
-	if( thermite )
-		if( istype(W, /obj/item/weapon/weldingtool) )
-			var/obj/item/weapon/weldingtool/WT = W
-			if( WT.remove_fuel(0,user) )
-				thermitemelt(user)
-				return
-
-		else if(istype(W, /obj/item/weapon/pickaxe/plasmacutter))
-			thermitemelt(user)
-			return
-
-		else if( istype(W, /obj/item/weapon/melee/energy/blade) )
-			var/obj/item/weapon/melee/energy/blade/EB = W
-
-			EB.spark_system.start()
-			user << "<span class='notice'>You slash \the [src] with \the [EB]; the thermite ignites!</span>"
-			playsound(src, "sparks", 50, 1)
-			playsound(src, 'sound/weapons/blade1.ogg', 50, 1)
-
-			thermitemelt(user)
+	if(thermite)
+		if(W.is_hot()) //HEY CAN THIS SET THE THERMITE ON FIRE ?
+			thermitemelt(user) //There, I just saved you fifty lines of redundant typechecks and awful snowflake coding
+			user.visible_message("<span class='warning'>[user] sets \the [src] ablaze with a swing of \his [W]</span>", \
+			"<span class='warning'>You set \the [src] ablaze with a swing of your [W]</span>")
 			return
 
-	else if(istype(W, /obj/item/weapon/melee/energy/blade))
-		user << "<span class='notice'>This wall is too thick to slice through. You will need to find a different path.</span>"
-		return
-
-	var/turf/T = user.loc	//get user's location for delay checks
-
-	//DECONSTRUCTION
+	//Deconstruction and reconstruction
 	switch(d_state)
 		if(0)
-			if (istype(W, /obj/item/weapon/wirecutters))
+			if(istype(W, /obj/item/weapon/wirecutters))
 				playsound(src, 'sound/items/Wirecutter.ogg', 100, 1)
 				src.d_state = 1
-				src.icon_state = "r_wall-1"
-				new /obj/item/stack/rods( src )
-				user << "<span class='notice'>You cut the outer grille.</span>"
+				update_icon()
+				var/obj/item/stack/rods/O = getFromPool(/obj/item/stack/rods, get_turf(src))

```

Comment:
You can just do getFromPool(/obj/item/stack/rods, get_turf(src), 2) since the args after the type are called for New().